junifer/preprocess/warping/space_warper.py

```diff
@@ -0,0 +1,192 @@
+        self, using: str, reference: str, on: Union[List[str], str]
+    ) -> None:
+        """Initialize the class."""
+        # Validate `using` parameter
+        valid_using = [dep["using"] for dep in self._CONDITIONAL_DEPENDENCIES]
+        if using not in valid_using:
+            raise_error(
+                f"Invalid value for `using`, should be one of: {valid_using}"
+            )
+        self.using = using
+        self.reference = reference
+        # Set required data types based on reference and
+        # initialize superclass
+        if self.reference == "T1w":
+            required_data_types = [self.reference, "Warp"]
+            # Listify on
+            if not isinstance(on, list):
+                on = [on]
+            # Extend required data types
+            required_data_types.extend(on)
+
+            super().__init__(
+                on=on,
+                required_data_types=required_data_types,
+            )
+        elif self.reference in tflow.templates():
+            super().__init__(on=on)
+        else:
+            raise_error(f"Unknown reference: {self.reference}")
+
+    def get_valid_inputs(self) -> List[str]:
+        """Get valid data types for input.
+
+        Returns
+        -------
+        list of str
+            The list of data types that can be used as input for this
+            preprocessor.
+
+        """
+        return ["T1w", "BOLD", "VBM_GM", "VBM_WM", "fALFF", "GCOR", "LCOR"]
+
+    def get_output_type(self, input_type: str) -> str:
+        """Get output type.
+
+        Parameters
+        ----------
+        input_type : str
+            The data type input to the preprocessor.
+
+        Returns
+        -------
+        str
+            The data type output by the preprocessor.
+
+        """
+        # Does not add any new keys
+        return input_type
+
+    def preprocess(
+        self,
+        input: Dict[str, Any],
+        extra_input: Optional[Dict[str, Any]] = None,
+    ) -> Tuple[Dict[str, Any], Optional[Dict[str, Dict[str, Any]]]]:
+        """Preprocess.
+
+        Parameters
+        ----------
+        input : dict
+            The input from the Junifer Data object.
+        extra_input : dict, optional
+            The other fields in the Junifer Data object.
+
+        Returns
+        -------
+        dict
+            The computed result as dictionary.
+        None
+            Extra "helper" data types as dictionary to add to the Junifer Data
+            object.
+
+        Raises
+        ------
+        ValueError
+            If ``extra_input`` is None when transforming to native space
+            i.e., using ``"T1w"`` as reference.
+        RuntimeError
+            If the data is in the correct space and does not require
+            warping.
+
+        """
+        logger.info(f"Warping to {self.reference} space using SpaceWarper")
+        # Transform to native space
+        if self.using in ["fsl", "ants"] and self.reference == "T1w":
```

Comment:
I'm missing one condition here (which should raise an error)

`using == "fsl"` and `reference != T1w"`